core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -1277,3 +1297,16 @@ trait ZPureLowPriorityImplicits {
         ZPure.tupledPar(fa, fb)
     }
 }
+
+trait ZPureSyntax {
+
+  implicit final class ZPureRefineToOrDieOps[W, S1, S2, R, E <: Throwable, A](self: ZPure[W, S1, S2, R, E, A]) {
+
+    /**
+     * Keeps some of the errors, and terminates the fiber with the rest.
+     */
+    def refineToOrDie[E1 <: E: ClassTag: CanFail](implicit ev: CanFail[E]): ZPure[W, S1, S2, R, E1, A] =
```

Comment:
I don't think we are actually preventing any mistakes here.

Reply:
This is what happens if I don't supply the type parameter when using `refineToOrDie`
```
[error] /home/ondra/Projects/zio-prelude/core/shared/src/test/scala/zio/prelude/fx/ZPureSpec.scala:793:54: This error handling operation assumes your effect can fail. However, your effect has Nothing for the error type, which means it cannot fail, so there is no need to handle the failure. To find out which method you can use instead of this operation, please see the reference chart at: https://zio.dev/docs/can_fail
[error]               assert(ZPure.attempt(a.toString.toInt).refineToOrDie.runEither)(
[error]                                                      ^
```

Reply:
Sorry, I meant by requiring this I don't think we are preventing any user errors. I haven't seen any cases of users getting confused by not supplying an argument to this method, and if they didn't it would be equivalent to `orDie` which is perfectly well defined. So I feel like we're not actually preventing user errors and we are preventing you from refining the error type to `Nothing` which, while there is a more specific operator for it, seems like something you should be able to do with the more general operator as well.

Reply:
IMHO it would be better to have a dedicated `orDie` and use of `refineToOrDie` with mandatory type argument, but if you're sure this is better :+1: